src/Microsoft.AspNetCore.Routing/Matchers/AmbiguousMatchException.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    /// <summary>
+    /// An exception which indicates multiple matches in endpoint selection.
+    /// </summary>
+    [Serializable]
+    internal class AmbiguousMatchException : Exception
```

Comment:
Why internal?

Reply:
I can't think of a case where a user would want to handle it. 

But it seems useful for us to have a specific exception type because the name you see in the debugger makes it pretty obvious that it's not a bug in the framework.